autoPyTorch/data/time_series_forecasting_validator.py

```diff
@@ -0,0 +1,353 @@
+# -*- encoding: utf-8 -*-
+from typing import Dict, Iterable, List, Optional, Tuple, Union
+
+
+import numpy as np
+
+import pandas as pd
+
+from sklearn.base import BaseEstimator
+from sklearn.exceptions import NotFittedError
+
+
+from autoPyTorch.data.tabular_validator import TabularInputValidator
+from autoPyTorch.data.time_series_feature_validator import TimeSeriesFeatureValidator, df2index
+from autoPyTorch.data.time_series_target_validator import TimeSeriesTargetValidator
+from autoPyTorch.data.utils import DatasetCompressionSpec
+
+
+class TimeSeriesForecastingInputValidator(TabularInputValidator):
+    """
+    A validator designed for a time series forecasting dataset.
+    As a time series forecasting dataset might contain several time sequence with different length, we will transform
+    all the data to DataFrameGroupBy whereas each group represents a series
+    TODO for multiple output: target names and shapes
+    """
+
+    def __init__(
+        self,
+        is_classification: bool = False,
+        logger_port: Optional[int] = None,
+        dataset_compression: Optional[DatasetCompressionSpec] = None,
+    ) -> None:
+        super(TimeSeriesForecastingInputValidator, self).__init__(
+            is_classification, logger_port, dataset_compression
+        )
+        self.feature_validator: TimeSeriesFeatureValidator = TimeSeriesFeatureValidator(
+            logger=self.logger
+        )
+        self.target_validator: TimeSeriesTargetValidator = TimeSeriesTargetValidator(
+            is_classification=self.is_classification, logger=self.logger
+        )
+        self._is_uni_variant = False
+        self.start_times: Optional[List[pd.DatetimeIndex]] = None
+        self.feature_shapes: Dict[str, int] = {}
+        self.feature_names: List[str] = []
+        self.series_idx: Optional[Union[List[Union[str, int]], str, int]] = None
+
+    def fit(  # type: ignore[override]
+        self,
+        X_train: Optional[Union[List, pd.DataFrame]],
+        y_train: Union[List, pd.DataFrame],
+        series_idx: Optional[Union[List[Union[str, int]], str, int]] = None,
+        X_test: Optional[Union[List, pd.DataFrame]] = None,
+        y_test: Optional[Union[List, pd.DataFrame]] = None,
+        start_times: Optional[List[pd.DatetimeIndex]] = None,
+    ) -> BaseEstimator:
+        """
+        fit the validator with the training data, (optionally) start times and other information
+        Args:
+            X_train (Optional[Union[List, pd.DataFrame]]): training features, could be None for "pure" forecasting tasks
```

Comment:
could you write the description in a new line with indentation? I think it messes with the documentation builder. [example](https://github.com/dengdifan/Auto-PyTorch/blob/4898ca5ef492b465ad1331df2b380b10be070087/autoPyTorch/data/base_feature_validator.py#L69)